libr/magic/apprentice.c

```diff
@@ -957,8 +957,8 @@ static int check_cond(RMagic *ms, int cond, ut32 cont_level) {
 /*
  * parse one line from magic file, put into magic[index++] if valid
  */
-static int parse(RMagic *ms, struct r_magic_entry **mentryp, ut32 *nmentryp, const char *line, size_t lineno, int action) {
-	static R_TH_LOCAL ut32 last_cont_level = 0;
+static int parse(RMagic *ms, struct r_magic_entry **mentryp, ut32 *nmentryp, const char *line, size_t lineno, int action, ut32 *last_cont_level) {
+	last_cont_level = 0;
```

Comment:
```suggestion
	*last_cont_level = 0;
```

Reply:
that change is incorrect as you are nullifying the pointer instead of the contents of it. but you need the contents of the pointer below. so it should be initialized before calling it not inside the function call